pkg/util/netops/netops.go

```diff
@@ -87,15 +87,19 @@ func GetNS() (*NS, error) {
 }
 
 // NewNamedNS returns an NS of the current network namespace
-func NewNamedNS(name string) (*NS, error) {
+func NewNamedNS(name string) (ns *NS, err error) {
 	runtime.LockOSThread()
 	defer runtime.LockOSThread()
 
 	old, err := netns.Get()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get the current network namespace: %w", err)
 	}
-	defer netns.Set(old)
+	defer func() {
+		if e := netns.Set(old); e != nil {
```

Comment:
How about using the same pattern as written here https://github.com/confidential-containers/cloud-api-adaptor/pull/43 ?

Reply:
@wainersm I think it should be fine.